README.md

```diff
@@ -1,171 +1,577 @@
+
+### Reflection-based seeding
 
-        var end = Random.DateTimeFrom(start);
-        
-        return new Holiday
+#### `Prerequisites` 
+When seeding without a class, we figure out the `Prerequisites` as follows:
+1. The library finds all fields that it thinks are navigation properties. This is based on the property being a `class`, and the existance of another property with the name `{PropertyName}Id`.
+2. A navigation property is considered as mandatory if its corresponding foreign key is not nullable.
+3. For each mandatory navigation property, we try to find a `EntitySeed<TNavigation>` class for it. If none is found, we'll repeat this process for `TNavigation`.
+
+#### `GetDefault` 
+When seeding without a class, the library will call the first constructor it finds for `TEntity`, using example values for any constructor arguments as needed:
+- if it is a `string`, a random `Guid` will be used.
+- otherwise, the default value for the type will be used.
+
+### Seeding method for each provider
+The following table shows how each provider behaves when seeding data: 
+
+| Provider | Model configuration | Reflection         |
+| --- |---------------|--------------------|
+| EF Core | ✔️            |  |
+| EF 6 |               | ️✔️️               |
+| In-memory |               | ️️️✔️              |
+
+## Custom `EntitySeed<TEntity>`s
+Creating a child class of `EntitySeed<TEntity>` class allows you define a valid default `TEntity` to be seeded looks like.
+It should seed _exactly_ the information required for an instance of `TEntity` to be considered valid. For example, it may be that you have two optional navigation properties, but your domain ensures exactly one is populated. This is a good use-case for a custom class as the default behaviour will be to seed neither of these navigation properties.
+
+You can create a class that inherits from `EntitySeed<TEntity>` to represent a single valid instance of `TEntity` as below: 
+```csharp
+public class BookingSeed : EntitySeed<Booking>
+{
+    public override IEnumerable<ISeedPrerequisite> Prerequisites() =>
+    [
+        new SeedPrerequisite<Booking, Facility>(f => f.Facility),
+        new SeedPrerequisite<Booking, Member>(f => f.Member)
+    ];
+
+    protected override Booking GetDefault(int index, Booking? previous)
+    {
+        return new Booking
         {
-            Start = start,
-            End = end,
-            Person = Existing<Person>().Random(),
-            Notes = Random.Sentence()
+            Notes = Guid.NewGuid().ToString()
         };
     }
 }
 ```
 
-Notice the usage of the `Previous` property to ensure that the holiday being generated is after the previous one.
+### When to override `Prerequisites` / `GetDfault`
+Both the `Prerequisites` and `GetDefault` methods are optional. If you don't need to explicitly state what this behaviour will be, the library will behave [as above](#seeding-without-a-class).
```

Comment:
Link doesn't work - there is no "Seeding without a class" heading.

Reply:
thanks - this is now "reflection-based-seeding"